packages/ag-charts-enterprise/src/features/error-bar/errorBar.ts

```diff
@@ -11,14 +11,56 @@ const {
     valueProperty,
     ChartAxisDirection,
     Validate,
-    NUMBER,
     OPT_BOOLEAN,
     OPT_COLOR_STRING,
     OPT_NUMBER,
     OPT_STRING,
 } = _ModuleSupport;
 
-type AnyCartesianSeries = _ModuleSupport.CartesianSeries<_Scene.Node, _ModuleSupport.CartesianSeriesNodeDatum>;
+type ErrorBoundCartesianSeries = Omit<
+    _ModuleSupport.CartesianSeries<
+        _Scene.Node,
+        _ModuleSupport.CartesianSeriesNodeDatum & _ModuleSupport.ErrorBoundSeriesNodeDatum
+    >,
+    'highlightSelection'
+>;
+type ModuleContext = _ModuleSupport.ModuleContext;
+type SeriesContext = _ModuleSupport.SeriesContext;
+
+function typeGuard<T extends ErrorBoundCartesianSeries>(
+    ctx: SeriesContext,
+    target: new (ctx: ModuleContext) => T
+): T | undefined {
+    if (target.prototype.type === ctx.series.type) {
+        return ctx.series as T;
+    }
+    return undefined;
+}
+
+function toErrorBoundCartesianSeries(ctx: SeriesContext): ErrorBoundCartesianSeries {
+    // This function checks that the requested ctx.series.type matches one of
+    // the static members `type` from the list of constructors below. If no
+    // match is found then an error is thrown. This ensures type safety because
+    // the compiler will ensure that the supported series types match the
+    // ErrorBoundCartesianSeries contract.
+    const supportedSeriesTypes: (new (ctx: ModuleContext) => ErrorBoundCartesianSeries)[] = [
+        _ModuleSupport.BarSeries,
+        _ModuleSupport.LineSeries,
+        _ModuleSupport.ScatterSeries,
+    ];
+    for (const ctor of supportedSeriesTypes) {
+        const series: ErrorBoundCartesianSeries | undefined = typeGuard(ctx, ctor);
+        if (series !== undefined) {
+            return series;
+        }
+    }
```

Comment:
I think it would be sufficient here to use the string literals `'bar', 'line', 'scatter'`, since under the covers in `typeGuard()` you are just doing a string comparison anyway - and avoid directly referencing/depending on the implementation types?